libsolidity/ast/Types.h

```diff
@@ -1067,6 +1097,16 @@ class TupleType: public Type
 
 protected:
 	std::vector<std::tuple<std::string, TypePointer>> makeStackItems() const override;
+	std::vector<Type const*> decomposition() const override
+	{
+		// Currently calling TupleType::decomposition() is not expected, because we cannot declare a variable of a tuple type.
+		// If that changes, before removing the solAssert, make sure the function does the right thing and is used properly.
+		// Note that different tuple members can have different data locations, so using decomposition() to check
+		// the tuple validity for a data location might require special care.
+		solAssert(false, "Tuple decomposition is not expected.");
```

Comment:
I think the unimplemented assert would make it clearer that it could have been implemented but just wasn't.
```suggestion
		solUnimplementedError(false, "Tuple decomposition is not expected.");
```